src/metabase/task/persist_refresh.clj

```diff
@@ -89,18 +89,20 @@
                                                 [:in (map :database_id deletables)]))
           unpersist-fn (fn []
                          (reduce (fn [stats persisted-info]
-                                   (let [database (-> persisted-info :database_id db-id->db)]
-                                     (log/info (trs "Unpersisting model with card-id {0}" (:card_id persisted-info)))
-                                     (try
-                                       (unpersist! refresher database persisted-info)
-                                       (db/delete! PersistedInfo :id (:id persisted-info))
-                                       (update stats :success inc)
-                                       (catch Exception e
-                                         (log/info e (trs "Error unpersisting model with card-id {0}" (:card_id persisted-info)))
-                                         (update stats :error inc)))))
+                                   ;; Since this could be long running, double check state just before deleting
+                                   (when (= "deletable" (db/select-one-field :state PersistedInfo :id (:id persisted-info)))
```

Comment:
nit: could destructure (:id persisted-info) and (:card_id persisted-info) since they're used 2x